### PR TITLE
Throwing custom AntiforgeryException for token validation failure sce…

### PR DESCRIPTION
**Pull request by ajaybhargavb**
…narios

Issue - #28 

@pranavkm @rynowak

### CODE REVIEW

File: src/Microsoft.AspNet.Antiforgery/AntiforgeryException.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="AntiforgeryException"/> with the specified
+        /// exception <paramref name="message"/>.
+        /// </summary>
+        /// <param name="message">The message that describes the error.</param>
+        public AntiforgeryException(string message)
+            : base(message)
+        {
+            if (message == null)
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Antiforgery/AntiforgeryException.cs]: You could remove the check. The base type should verify it for you.

File: src/Microsoft.AspNet.Antiforgery/DefaultAntiforgeryTokenGenerator.cs [hunk 2]
```diff
@@ -153,20 +153,20 @@ public bool IsCookieTokenValid(AntiforgeryToken cookieToken)
 
             if (!comparer.Equals(requestToken.Username, currentUsername))
             {
-                throw new InvalidOperationException(
+                throw new AntiforgeryException(
                     Resources.FormatAntiforgeryToken_UsernameMismatch(requestToken.Username, currentUsername));
             }
 
             if (!Equals(requestToken.ClaimUid, currentClaimUid))
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Antiforgery/DefaultAntiforgeryTokenGenerator.cs]: Could you change this to `object.Equals`

File: src/Microsoft.AspNet.Antiforgery/AntiforgeryException.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Antiforgery
+{
+    /// <summary>
+    /// The <see cref="Exception"/> that is thrown when the antiforgery token validation fails.
+    /// </summary>
+    public class AntiforgeryException : Exception
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Antiforgery/AntiforgeryException.cs]: `AntiforgeryValidationException`?

Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Antiforgery/AntiforgeryException.cs]: Sounds better. :+1:

**Comment by ajaybhargavb** [comment]
Updated.

**Comment by rynowak** [comment]
:shipit: from me

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**